src/morphology.hpp

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+// Representation of 3-d embedded cable morphology, independent of other
+// cell information.
+
+#include <vector>
+
+namespace nest {
+namespace mc {
+
+struct section_point {
+    double x, y, z, r;  // [µm], r is radius.
+};
+
+enum class section_kind {
+    soma,
+    dendrite,
+    axon,
+    none
+};
+
+struct section_geometry {
```

Comment:
see my comments above. If this were a class, you could have an `add_point` or `add_points` member that ensures that `length` is always up to date.